modules/ROOT/pages/http/http-authentication.adoc

```diff
@@ -2,357 +2,496 @@
 
-[.ex]
-=====
-[discrete.view]
-=== Studio Visual Editor
+=== OAuth 2 Authorization - Studio Visual Editor
 
-. Add the OAuth Module from the palette.
-. In Studio, select the HTTP Request Configuration global element where you want to use the OAuth authorization code grant type.
-. On the *Authentication* combo box, select `Authorization code grant type`.
-. Set the following options:
+. Drag an *HTTP* > *Request* operation from the Mule Palette to the Process area of the Studio flow. This operation is known as the HTTP Request connector.
+. Expand the Package Explorer and double-click the `pom.xml` file.
+. At the end of the `<dependencies>` section and before the `</dependencies>` statement,
+add these statements to enable OAuth options for the HTTP Request connector:
 +
-* Client Id: Type the client Id that Github provided when you registered the app.
-* Client Secret: Type the client secret that Github provided when you registered the app.
-* Local callback URL: `+http://localhost:8082/callback+`
+[[setpomfile]]
+[source,xml,linenums]
+----
+<dependency>
+    <groupId>org.mule.modules</groupId>
+    <artifactId>mule-oauth-module</artifactId>
+    <version>1.1.2</version>
+    <classifier>mule-plugin</classifier>
+</dependency>
+----
++
+. In the properties editor for *Connector Configuration*, click the green plus icon (or click the Edit icon if you have already created a configuration).
+. Set the *Authentication* field to *Authorization code grant type*.
+. Set the following required fields:
++
+** *External callback url*: `+http://localhost:8082/callback+`
++
+The OAuth authentication server uses this URL to provide the authentication code to the Mule server so that the Mule server can retrieve the access token. This must be the externally visible address of the callback, not the local one.
++
+** *Local authorization url*: `+https://localhost:8082/login+`
++
+Lets you authenticate and grant access to the app for your account.
++
+** *Authorization url*: `+https://github.com/login/oauth/authorize+`
++
+Authorize the app for a user.
++
+** *Client id*: Type the client ID that GitHub provided when you registered the app.
+** *Client secret*: Type the client secret that GitHub provided when you registered the app.
+** Token url: `+https://github.com/login/oauth/access_token+`
++
+You can also set these optional fields:
 +
-This value matches the value you configured for *Authorization callback URL* when registering the app in Github.
-* Authorization Url: `+https://github.com/login/oauth/authorize+`
-* Local Authorization Url: `+https://localhost:8082/login+`
-* Token Url: `+https://github.com/login/oauth/access_token+`
-* Response Access Token: `#[payload.'access_token']`
+** *Local callback url*: `+http://localhost:8082/callback+`
++
+This value matches the value you configured for *Authorization callback URL* when registering the app in GitHub.
```

Comment:
Authorization callback URL -> External callback url

Reply:
This is the configuration of the server that mule will create to receive the requests that a remote host sends to the externalCallbackUrl. external and internal callback url are the same, just that one is the configuration to create a server in the runtime (internal) and the other is how that server is seen from the internet (external). This should be explained here in the docs.

Reply:
Fixed